README.md

```diff
@@ -2,365 +2,402 @@
+be sent one after another, ignoring any flush timer.
 
-* _**Optional**_
-* Type: `String`
-* Default: `''`
-* Values: `10.0.0.1`
 
-The default IP Address passed along with every log sent through this instance.
+### Event: `'warn'`
+* [`<Object>`][]
+    * `message` [`<String>`][] - The warn message.  Depends on context.
 
-##### level
+This is emitted when there is no log data provided to the `log` method, or when `removeMetaProperty` is called with an unrecognized key.  
+For those cases, additional properties (apart from `message`) are included:
 
-* _**Optional**_
-* Type: `String`
-* Default: `Info`
-* Values: `Debug`, `Trace`, `Info`, `Warn`, `Error`, `Fatal`, `YourCustomLevel`
-* Max Length: `32`
+**Warnings during `log()`**
 
-The default level passed along with every log sent through this instance.
+* `statement` (Any) - If `log()` was called with a `null` string, or an invalid data type, this will contain the given log statement
 
-##### mac
+**Warnings during `removeMetaProperty`**
 
-* _**Optional**_
-* Type: `String`
-* Default: `''`
-* Values: `C0:FF:EE:C0:FF:EE`
+* `key` [`<String>`][] - The key that was attempted to be removed, but did not exist
 
-The default MAC Address passed along with every log sent through this instance.
+## API
 
-##### max_length
+### `createLogger(key[, options])`
+
+* `key` [`<String>`][] - Your [ingestion key](https://docs.logdna.com/docs/ingestion-key)
+* `options` [`<Object>`][]
+    * `level` [`<String>`][] - Level to be used if not specified elsewhere. **Default:** `INFO`
+    * `tags` [`<Array>`][] | [`<String>`][] - Tags to be added to each message
+    * `meta` [`<Object>`][] - Global metadata.  Added to each message, unless overridden.
+    * `timeout` [`<Number>`][] - Millisecond timeout for each HTTP request. **Default:** `5000`. **Max:** `300000`
+    * `hostname` [`<String>`][] - Hostname for each HTTP request.
+    * `mac` [`<String>`][] - MAC address for each HTTP request.
+    * `ip` [`<String>`][] - IPv4 or IPv6 address for each HTTP request.
+    * `url` [`<String>`][] - URL of the logging server.  **Default:** `https://logs.logdna.com/logs/ingest`
+    * `flushLimit` [`<Number>`][] - Maximum total line lengths before a `flush` is forced. **Default:** `5000000`
+    * `flushInterval` [`<Number>`][] - Milliseconds to wait before sending the buffer. **Default:** `250`
+    * `shimProperties` [`<Array>`][] - List of dynamic `options` keys to look for when calling `log()`
+    * `indexMeta` [`<Boolean>`][] - Controls whether or not `meta` data for each message is searchable. **Default:** `false`
+    * `app` [`<String>`][] - Arbitrary app name for labeling each message. **Default:** `default`
+    * `env` [`<String>`][] - An environment label attached to each message
+    * `baseBackoffMillis` [`<Number>`][] - Minimum exponential backoff time. **Default:** `3000`
+    * `maxBackoffMillis` [`<Number>`][] - Maximum exponential backoff time. **Default:** `30000`
```

Comment:
Well, it's in the name for these.  Is "Millis" not as descriptive as `ms` ?  I guess I thought that it would be, but maybe not.  We can also put `ms` in those descriptions.

Reply:
sort of just thinking in terms of aligning conventions here and internally, I think it's clear either way so I'll leave it to you

Reply:
I'm down for consistency for all of these.  For example `flushInterval` should be `flushIntervalMs`.  That work for you?